x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/results/ClassificationInferenceResults.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.ml.inference.results;
+
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.xpack.core.ml.utils.ExceptionsHelper;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+public class ClassificationInferenceResults extends SingleValueInferenceResults {
+
+    public static final String RESULT_TYPE = "classification";
+    public static final ParseField CLASSIFICATION_LABEL = new ParseField("classification_label");
+    public static final ParseField TOP_CLASSES = new ParseField("top_classes");
+    
+    private final String classificationLabel;
+    private final List<TopClassEntry> topClasses;
+
+    public ClassificationInferenceResults(double value, String classificationLabel, List<TopClassEntry> topClasses) {
+        super(value);
+        this.classificationLabel = classificationLabel;
+        this.topClasses = topClasses == null ? null : Collections.unmodifiableList(topClasses);
```

Comment:
Could we ensure `topClasses` is never `null`? It'd save us the if statements for the read/write methods. Also, when turning to XContent, we can check the list is not empty to decide whether to include the field.